exp/internal/controllers/machinepool_controller_phases.go

```diff
@@ -492,6 +455,11 @@ func getNewMachine(mp *expv1.MachinePool, infraMachine *unstructured.Unstructure
 		},
 	}
 
+	if existingMachine != nil {
+		machine.SetName(existingMachine.Name)
```

Comment:
What is the scenario when the name will have to be updated?

Reply:
We aren't updating the name, but just making sure the new desired machine state has the same name as the existing Machine for patching. It's copied from the [MachineSet controller](https://github.com/kubernetes-sigs/cluster-api/blob/34a990cc868f22044cc9dc897798b41023033861/internal/controllers/machineset/machineset_controller.go#L628-L633).